espree.js

```diff
@@ -3857,6 +3858,15 @@ function parseParams(firstRestricted) {
                     message = Messages.StrictParamDupe;
                 }
             }
+            if (match("=")) {
+                if (!allowDefaultParams) {
+                    throwUnexpected(lookahead);
+                }
+                lex();
+                def = parseAssignmentExpression();
+                ++defaultCount;
+            }
+            defaults.push(def);
```

Comment:
Also keep in mind that Esprima does have various bugs with ES6 features that haven't been fixed. Just because it's in Esprima doesn't mean it's correct.